frontend/src/components/Auth/AuthContext.test.js

```diff
@@ -0,0 +1,78 @@
+import React, { useContext } from 'react';
+import { render, act, screen, cleanup } from '@testing-library/react';
+import { AuthContext, AuthProvider } from './AuthContext.js';
+
+jest.useFakeTimers();
+
+// Mock the the Firebase Auth onAuthStateChanged function, which pauses for 1
+// second before returning a fake user with only a name field set.
+const mockOnAuthStateChanged = jest.fn(callback => {
+  setTimeout(() => {
+    callback({ name: 'Keiffer' })
+  }, 1000);
+});
+jest.mock('firebase/app', () => {
+  return {
+    initializeApp: () => {
+      return {
+        auth: () => {
+          return {
+            onAuthStateChanged: mockOnAuthStateChanged
+          }
+        }
+      }
+    }
+  }
+});
+
+describe('AuthProvider component', () => {
+  beforeEach(() => { render(<AuthProvider />) });
+
+  afterEach(cleanup);
+
+  it('initially displays "Loading"', () => {
+    act(() => jest.advanceTimersByTime(500));
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+  });
+
+  it('returns a provider when onAuthStateChanged is called', () => {
+    act(() => jest.advanceTimersByTime(2000));
+    expect(screen.queryByText('Loading...')).not.toBeInTheDocument();
+  });
+});
+
+describe('AuthContext Consumer component', () => {
+  // A Consumer component for AuthContext that just displays the current
+  // user.
+  const TestAuthContextConsumerComponent = () => {
+    const currentUser = useContext(AuthContext);
+
+    return (
+      <div>
+        <div>{currentUser.name}</div>
+      </div>
+    );
+  };
+
+  beforeEach(() => {
+    render(
+      <AuthProvider>
+        <TestAuthContextConsumerComponent />
+      </AuthProvider>
+    );
+  });
```

Comment:
Looks like both `describe`s have similar beforeEachs(). Can they just be a top-level beforeach()

Reply:
I don't think I'll be able to change the `beforeEach`s since they're both testing very slightly different things. I did move the `afterEach` to the top level though as that was being repeated between the `describe`s.